src/Spago/Command/Build.purs

```diff
@@ -202,7 +207,7 @@ getBuildGlobs { selected, dependencies, withTests, depsOnly } =
   dependencyGlobs = (Tuple.uncurry $ Config.sourceGlob NoTestGlobs) =<< dependencyPkgs
   monorepoPkgGlobs
     | depsOnly = []
-    | otherwise = (Tuple.uncurry $ Config.sourceGlob testGlobs) =<< monorepoPkgs
+    | otherwise = (Tuple.uncurry $ Config.sourceGlob NoTestGlobs) =<< monorepoPkgs
```

Comment:
This is explained in the description: 

* `Build.getBuildGlobs` no longer adds test globs (i.e. `test/**/*.purs`) for all workspace packages, but only for the "selected" ones. This is because only "selected" packages will have their test dependencies added to the build plan.

Reply:
Yeah I read that, but I'm still confused and worried that things will subtly break. Let's move on with this and see what happens

Reply:
Maybe an example would help.

```
package:
  name: a
  test: { dependencies: [ x, y ] }
package:
  name: b
  test: { dependencies: [ p, q, a ] }
```

Note that `b`'s tests use `a` in some way. Perhaps `a` is a shared library with some cool utility functions that are useful in `b`'s tests.

When we run `spago test -p b`, we have to build `p`, `q`, and `a`, because, well, `b`'s tests depend on them.

Question: should we also build `x` and `y`?

My answer is "no". They are used in `a`'s tests, not in `a` itself, and it's only `a` itself that is used in `b`'s tests.